helios-client/src/main/java/com/spotify/helios/client/DefaultRequestDispatcher.java

```diff
@@ -243,7 +243,7 @@ private HttpURLConnection connect(final URI uri, final String method, final byte
 
               final int responseCode = connection.getResponseCode();
               if (((responseCode == HTTP_FORBIDDEN) || (responseCode == HTTP_UNAUTHORIZED))
-                  && !identities.isEmpty()) {
+                  && connection.getHeaderField("WWW-Authenticate") != null) {
```

Comment:
@rohansingh We need some way to differentiate between app-level and TLS-level unauthed responses. Would a header set by nginx work?